pkg/manifest/manifest.go

```diff
@@ -15,25 +17,60 @@ import (
 	"k8s.io/client-go/kubernetes/scheme"
 )
 
+// ResourceIdYaml is used to parse out the information
+// that uniquely identifies a Kubernetes resource.
+type ResourceIdYaml struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Metadata   struct {
+		Name      string `yaml:"name"`
+		Namespace string `yaml:"namespace"`
+	}
+}
+
+// ResourceId uniquely identifies a Kubernetes resource.
+// It is used to identify any duplicate resources within
+// a given set of manifests.
+type ResourceId struct {
+	Group     string
+	Kind      string
+	Name      string
+	Namespace string
```

Comment:
nit: copy a sentence or two of godocs for these properties from [`ObjectReference`][1] or similar?

[1]: https://pkg.go.dev/k8s.io/api/core/v1#ObjectReference

Reply:
How about this?